### PR TITLE
print value of memory pointer when no arguments are given

### PR DESCRIPTION
**Pull request by vlaube-de**
as requested in #11

### CODE REVIEW

File: kernel/shell.asm [hunk 1]
```diff
@@ -220,20 +220,27 @@ shellPrintErr:
...
 ; Example: mptr 01fe
 shellMptrCmd:
-	.db	"mptr", 0b011, 0b001, 0
+	.db	"mptr", 0b111, 0b001, 0
 shellMptr:
 	push	hl
 
+	ld	a, (hl)
+	inc	hl
+	and (hl)
```
Comment (dprentiss) [inline kernel/shell.asm]: I think this should be:  OR (HL)

**Comment by hsoft** [comment]
One of the (many) rough edges in Collapse OS is that argument parsing can't differentiate between a `0` argument and no argument. So what should be an easy task in fact isn't...

File: kernel/shell.asm (hunk 1 above)
Comment (dprentiss) [reply kernel/shell.asm]: OR doesn't work either. It is not possible to change the pointer to 0000.
Eg:
```
> mptr 0000
FF00

**Comment by vlaube-de** [comment]
This should do it.

**Comment by vlaube-de** [comment]
The diff is hard to read because I'm undoing most of the earlier changes, sorry.

**Comment by hsoft** [comment]
Thanks for your work, but I'm not inclined to merge this. The "mptr without argument prints it" idea isn't bad because even if it slightly useless (you'll see with usage that you rarely forget where your `mptr` was last set. It's a command you often use), it makes sense and is light.

If we have to add a new command, then it's not worth it.

What is actually needed is to improve the argument parsing routine to differentiate 0 from absent, but that's some significant work that require a bit of familiarity with the shell code as a whole.

I appreciate the gesture though, your code looks good.

**Comment by dprentiss** [comment]
Hacking at the shell is definitely beyond my assembly skill. But, now I have a good project for learning to add a new command in glue code. If I understood the general philosophy of this project, adding user space functions in glue code is the preferred approach anyway.

**Pull request closed without merging**